### PR TITLE
Add flatShading prop to ColumnLayer

### PR DESCRIPTION
**Pull request by Pessimistress**
For #6366 

#### Change List
- Add flatShading prop to ColumnLayer
- H3HexagonLayer renders ColumnLayer with `flatShading: true`
- Unit test
- Render tests
- Docs

### CODE REVIEW

**Review by ibgreen — APPROVED** [review]
Got to say that this looks quite complicated, I am not going to try to parse it. 

flatShading is normally just a few lines of work, adjusting the vertex normals for each face from face orthogonal to average with neighbor face, then relying on shader interpolation. Wondering if there are any improvements (such as oct-encoded normals) that would let us make this code less custom?

File: modules/layers/src/column-layer/column-geometry.js [hunk 1]
```diff
@@ -5,117 +5,137 @@ import {modifyPolygonWindingDirection, WINDING} from '@math.gl/polygon';
...
+      for (let k = 0; k < verticesPerLocation; k++) {
+        positions[pi + 0] = vertices[j * 2];
+        positions[pi + 1] = vertices[j * 2 + 1];
+        positions[pi + 2] = (1 / 2 - (k % 2)) * height;
+
+        if (flat) {
+          const leftJ = k < 2 ? prevJ : j;
+          const rightJ = k < 2 ? j : nextJ;
+          normals[pi + 0] = vertices[rightJ * 2 + 1] - vertices[leftJ * 2 + 1]; // dy
+          normals[pi + 1] = vertices[leftJ * 2] - vertices[rightJ * 2]; // -dx
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/layers/src/column-layer/column-geometry.js:103]: They are normalized in the project shader (because some transforms is usually required).

**Comment by Pessimistress** [comment]
@ibgreen you are describing how the layer works today - all normals are average between neighbor faces. To add flat shading we need to double up the vertices in order to assign them different normals for each face.

The ColumnLayer uses one geometry to render all of:
- The cap, either from a user-supplied vertex array or a regular polygon
- The extrusion, may be smooth shading (shared vertices) or flat shading (separate vertices)
- The stroke around the polygon, if `extruded: false`
- The wireframe of the column, if `extruded: true`

**Comment by ibgreen** [comment]
Got it, that makes sense.

BTW, didn't we add non-geometry dependent flatshading using shader derivatives in some place?

**Comment by Pessimistress** [comment]
Yes, in the SimpleMeshLayer. All core layers use gouraud shading because it's cheaper.

**Comment by ibgreen** [comment]
But the way normal calculation is done should be independent of shading model.

**Review by ibgreen — APPROVED** [review]
Regarding when can use derivatives to calculate normals and when we need to generate more verts to add pre-calculated normals:

As it stands this PR now always generate more verts, even if using fragment lighting? Which is probably fine as presumably fragment lighting adds enough overhead that it is not worth reducing verts in that case?

**Comment by Pessimistress** [comment]
I tested adding vertices vs. derivatives and there isn't a measurable difference. I've updated this PR to use the derivatives approach for simplicity.

**Pull request merged**